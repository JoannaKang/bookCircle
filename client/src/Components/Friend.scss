$primary-color: #DA291C;
$primary-dark: #1C191A;
$background-color: #F6E7D8;

.friendDiv {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 2rem;
}

.friendIconAndNameDiv {
  display: flex;
  align-items: center;

  .friendIcon {
    height: 70px;
    width: 70px;
    color: $primary-dark;
  }

  h3 {
    margin-left: 0.5rem;
    font-size: 1.8rem;
    font-weight: 400;
    color: $primary-dark;
  }
}

.friendButton {
  background-color: $primary-dark;
  color: $background-color;
  font-size: 1.2rem;
  border: none;
  border-radius: 5px;
  padding: 0.5rem;
  height: 0%;
}