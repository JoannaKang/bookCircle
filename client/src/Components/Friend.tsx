import React, { useEffect, useState } from "react"
import { useHistory } from "react-router-dom"; 
import { BsFillPersonFill } from "react-icons/bs";
import {getFriendName} from '../ApiService/serverApiService'
import './Friend.scss'

interface myProps {
  friendId: string;
  getSelectedFriend: Function;
}

// props.history + push to the history.push
const Friend = ({friendId,  getSelectedFriend}: myProps) => {
  const [friendName, setFriendName] = useState('')
  let history = useHistory();

  useEffect(() => {
    getFriendsName(friendId)
  },[friendId])

  async function getFriendsName(id: string) {
    const result:any = await getFriendName(id)
    setFriendName(result);
  }

  async function handleClick() {
    await getSelectedFriend(friendName)
    history.push("/friendsLibrary")
  }

  return (
    <div className="friendDiv">
      <div className="friendIconAndNameDiv">
        <BsFillPersonFill className="friendIcon" />
        {<h3>{friendName}</h3>}
      </div>
        <button className="friendButton" onClick={handleClick} data-testid="friendButtonClick">
          view books
        </button>    
    </div>
  )
}

export default Friend;