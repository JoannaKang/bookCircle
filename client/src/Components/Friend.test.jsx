import Friend from './Friend'
import { render, fireEvent, getByTestId } from '@testing-library/react'

describe('Friend', () => {

    let name = "Edward"; 

    it('matches snapshot when friendID is provided', () => {
        const { container } = render(<Friend friendId={name}></Friend>)
        expect(container).toMatchSnapshot()
    });
    
    it('HandleClick called when Friend Button is clicked', () => {
        const getSelectedFriend = jest.fn();
        const { getByTestId } = render(<Friend friendId={name} getSelectedFriend={getSelectedFriend}></Friend>)
  
        fireEvent.click(getByTestId("friendButtonClick"))
        expect(getSelectedFriend)

    });

})

