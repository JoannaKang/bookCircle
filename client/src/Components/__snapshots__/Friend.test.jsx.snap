// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friend matches snapshot when friendID is provided 1`] = `
<div>
  <div
    class="friendDiv"
  >
    <div
      class="friendIconAndNameDiv"
    >
      <svg
        class="friendIcon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 100-6 3 3 0 000 6z"
          fill-rule="evenodd"
        />
      </svg>
      <h3 />
    </div>
    <button
      class="friendButton"
      data-testid="friendButtonClick"
    >
      view books
    </button>
  </div>
</div>
`;
